src/osd/PrimaryLogPG.cc

```diff
@@ -11018,22 +11018,22 @@ bool PrimaryLogPG::start_recovery_ops(
   int unfound = get_num_unfound();
   if (unfound) {
     dout(10) << " still have " << unfound << " unfound" << dendl;
-    return work_in_progress;
+    return !work_in_progress;
```

Comment:
if `work_in_progress` is true, we will return at https://github.com/ceph/ceph/pull/17708/commits/2b432c1199652887b0a662b4f0344e208304f165#diff-fb41013d27e932534adb50eb3de2aaa5R11007, so `!work_in_progress` always evaluates to `true` here.

Reply:
@liewegas not addressed

Reply:
it's fixed in ea32e82e6e4a52ef4c6728d36dfc747b053c5edb, not sure why github is still showing this